File: datagym-frontend/src/app/label-config/model/geometry/LcEntryPolygon.ts
import { LcEntryGeometry } from './LcEntryGeometry';
import { LcEntryType } from '../LcEntryType';

export class LcEntryPolygon extends LcEntryGeometry {
  static type: LcEntryType = LcEntryType.POLYGON;

  constructor(lcEntryParentId: string | null, key?: string, value?: string, color?: string, shortcut?: string) {
    super(LcEntryPolygon.type, lcEntryParentId, key, value, color, shortcut);
  }
}

File: datagym-backend/src/main/java/ai/datagym/application/labelConfiguration/entity/geometry/LcEntryPolygon.java
package ai.datagym.application.labelConfiguration.entity.geometry;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import java.io.Serializable;

@Entity
@DiscriminatorValue(value = "polygon")
public class LcEntryPolygon extends LcEntryGeometry implements Serializable {
    public LcEntryPolygon() {
        super();
    }
}
